python/lib/dependabot/python/file_updater/pipfile_preparer.rb

```diff
@@ -130,9 +130,15 @@ def sub_auth_url(source, credentials)
 
         def config_variable_sources(credentials)
           @config_variable_sources ||=
-            credentials.
-            select { |cred| cred["type"] == "python_index" }.
-            map { |c| { "url" => AuthedUrlBuilder.authed_url(credential: c) } }
+            credentials.select { |cred| cred["type"] == "python_index" }.
+            map do |c|
+              {
+                "url" => AuthedUrlBuilder.authed_url(credential: c),
+                # Random suffix ensure names are unique. This entire source is stripped out of the final Pipfile.lock in
+                # FileUpdater#post_process_lockfile so it's okay that the names are non-deterministic.
```

Comment:
Actually lemme add a spec that ensures this is stripped out of the final `Pipfile[.lock]` files